packages/server/modules/workspaces/graph/resolvers/workspaces.ts

```diff
@@ -740,6 +756,59 @@ export = FF_WORKSPACES_MODULE_ENABLED
         }
       },
       WorkspaceProjectMutations: {
+        create: async (_parent, args, context) => {
+          const rateLimitResult = await getRateLimitResult(
+            'STREAM_CREATE',
+            context.userId!
+          )
+          if (isRateLimitBreached(rateLimitResult)) {
+            throw new RateLimitError(rateLimitResult)
+          }
+
+          await authorizeResolver(
+            context.userId!,
+            args.input.workspaceId,
+            Roles.Workspace.Member,
+            context.resourceAccessRules
+          )
+
+          // TODO: resource access rules check
+
+          // TODO: get workspace's region here
+          const regionKey = 'region1'
+
+          const projectDb = await getDb({ regionKey })
+
+          const storeModelFactory =
+            ({ db }: { db: Knex }): StoreModel =>
+            async ({ authorId, projectId, name, description }) => {
+              await createBranchFactory({ db })({
+                authorId,
+                description,
+                name,
+                streamId: projectId
+              })
+            }
+
+          // todo, use the command factory here, but for that, we need to migrate to the event bus
+          const createNewProject = createNewProjectFactory({
```

Comment:
Can't say I'm a fan of the DB selection logic being left up to the dev writing the DI code. You can accidentally use the wrong DB here and fuck shit up for real. I guess if we get a DI container in place its going to be better.

I guess you could also build a factory w/ just 2 deps - db and project db, and then that factory could correctly build createNewProject and all related factories.

Reply:
guys, lets not optimize for things before we know the pattern.

This is the first go at a multi region service instantiation. I do not know, what will work, but from the prev trials with the DI container, this is very possible to be centralized and codified in one place.

Until  then, I'm going to type these out to see what works and what doesn't